homeassistant/components/yeelight/config_flow.py

```diff
@@ -0,0 +1,249 @@
+        self._capabilities = capabilities
+        await self.async_set_unique_id(capabilities["id"])
+        self._abort_if_unique_id_configured()
+
+        user_input[CONF_DISCOVERY] = False
+        return self.async_create_entry(
+            title=user_input[CONF_IP_ADDRESS], data=user_input
+        )
+
+    @callback
+    def _async_ip_already_configured(self):
+        """See if we already have an endpoint matching user input."""
+        for entry in self._async_current_entries():
+            if entry.data.get(CONF_IP_ADDRESS) == self._ipaddr:
+                return True
+        return False
+
+
+@callback
+def _async_options_data_schema(options: dict) -> vol.Schema:
+    return vol.Schema(
+        {
+            vol.Optional(CONF_MODEL, default=options[CONF_MODEL],): str,
+            vol.Required(
+                CONF_TRANSITION, default=options[CONF_TRANSITION],
+            ): cv.positive_int,
+            vol.Required(CONF_MODE_MUSIC, default=options[CONF_MODE_MUSIC],): bool,
+            vol.Required(
+                CONF_SAVE_ON_CHANGE, default=options[CONF_SAVE_ON_CHANGE],
+            ): bool,
+            vol.Required(
+                CONF_NIGHTLIGHT_SWITCH, default=options[CONF_NIGHTLIGHT_SWITCH],
+            ): bool,
+        }
+    )
+
+
+class ManualOptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle a option flow for manually set Yeelight."""
+
+    def __init__(self, config_entry):
+        """Initialize the option flow."""
+        self._config_entry = config_entry
+
+    async def async_step_init(self, user_input=None):
+        """Handle the initial step."""
+        return await self.async_step_options()
+
+    async def async_step_options(self, user_input=None):
+        """Handle the step to change options."""
+        if user_input is not None:
+            return self.async_create_entry(
+                title=self._config_entry.data[CONF_IP_ADDRESS], data=user_input
+            )
+
+        return self.async_show_form(
+            step_id="options",
+            data_schema=_async_options_data_schema(self._config_entry.options),
+        )
+
+
+class DiscoveryOptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle a option flow for discovery Yeelight."""
+
+    def __init__(self, config_entry):
+        """Initialize the option flow."""
+        self._config_entry = config_entry
+        self._devices = None
+        self._picked_device = None
+
+    async def async_step_init(self, user_input=None):
+        """Handle option flow initialization."""
+        device_registry = await dr.async_get_registry(self.hass)
+        self._devices = {}
+        for unique_id in self._config_entry.options:
+            device = device_registry.async_get_device(
+                identifiers={(DOMAIN, unique_id)}, connections={}
+            )
+            name = device.name_by_user or device.name
+            self._devices[f"{name} ({unique_id})"] = unique_id
```

Comment:
If we invert this dictionary and pass the whole dictionary to `vol.In` the values will be shown to the user and the selected item key will be returned as user input.